File: mpp_proiect_java/src/main/java/triatlon/domain/Participant.java
package triatlon.domain;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

public class Participant extends Entity<Long> {

    private String firstName;
    private String lastName;
    private List<Rezultat> rezultate;
    private Integer varsta;
    private Integer totalPuncte;

    public Participant(String firstName, String lastName) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.rezultate = new ArrayList<Rezultat>();
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }


  @Override
    public String toString() {
        return "Participant{" +
                "firstName='" + firstName + '\'' +
                ", lastName='" + lastName + '\'' +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Participant)) return false;
        Participant that = (Participant) o;
        return getFirstName().equals(that.getFirstName()) &&
                getLastName().equals(that.getLastName()) &&
                getId().equals(that.getId());
    }

    @Override
    public int hashCode() {
        return Objects.hash(getId(),getFirstName(), getLastName());
    }
}

File: mpp_proiect_CSharp/triatlon/triatlon/domain/Participant.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace triatlon.domain
{
    class Participant : Entity<long>
    {
        public long Id { get; set; }
        public string firstName { get; set; }
        public string lastName { get; set; }
        public int varsta { get; set;}

        public Participant(long id, string firstname, string lastname,int varstaP)
        {
            this.Id = id;
            this.firstName = firstname;
            this.lastName = lastname;
            this.varsta = varstaP;
        }
        public override string ToString()
        {
            return "Participant - nume: " + firstName + '|' +
                "prenume=" + lastName;
        }
    }
}
